src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java

```diff
@@ -192,36 +201,40 @@ private static List<String> createCorpusList(String filePath, String fileType, S
     return corpus;
   }
 
-  private static double calculateCosineSimilarity(JSONArray vec1, JSONArray vec2) {
-    double dotProduct = 0.0;
-    double normA = 0.0;
-    double normB = 0.0;
-    for (int i = 0; i < vec1.length(); i++) {
-      double a = vec1.getDouble(i);
-      double b = vec2.getDouble(i);
-      dotProduct += a * b;
-      normA += Math.pow(a, 2);
-      normB += Math.pow(b, 2);
+  private static BigDecimal calculateCosineSimilarity(List<BigDecimal> embeddingList, List<BigDecimal> corpusEmbedding) {
```

Comment:
I don't think, we need BigDecimal here. The generated embeddings are  0.000002239895, -0.0038747797, 0.008896385 etc. The double can accurately handle the magnitude and precision without losing information. There may be minimal floating-point errors but the cumulative error in a typical cosine similarity calculation is unlikely to affect. BigDecimal would significantly slow down computations compared to double. Cosine similarity involves multiple multiplications, additions, and a square root, all of which are computationally heavier with BigDecimal.

Reply:
Done. Changed to double